control-plane/connect-inject/controllers/pod/pod_controller.go

```diff
@@ -664,10 +664,16 @@ func parseLocality(node corev1.Node) *pbcatalog.Locality {
 
 func metaFromPod(pod corev1.Pod) map[string]string {
 	// TODO: allow custom workload metadata
-	return map[string]string{
+	meta := map[string]string{
 		constants.MetaKeyKubeNS:    pod.GetNamespace(),
 		constants.MetaKeyManagedBy: constants.ManagedByPodValue,
 	}
+
+	if gatewayKind := pod.Annotations[constants.AnnotationGatewayKind]; gatewayKind != "" {
```

Comment:
should this be using the comma ok idiom to guard against the key not existing in the map?

Reply:
not necessary since `gatewayKind` will be an empty string in that case as well